syntax = "proto3";

package usersystem;

//Message for user details

message User {
    string username = 1;
    string password = 2;
    string email = 3;
    int32 age = 4;
}

message RegisterRequest {
    User user = 1;
}

message RegisterResponse {
    bool success = 1;
    string message = 2;
}

service UserRegistration {
    rpc RegisterUser(RegisterRequest) returns (RegisterResponse);
}